src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/PartBRecordComponentTest.kt

```diff
@@ -67,22 +68,50 @@ class PartBRecordComponentTest : ComponentTestBase() {
     assertThat(recallWithPartBRecord.partBRecords.first().version, equalTo(1))
   }
 
-  // TODO PUD-1605 Multiple error handling for virus found
-
   @Test
-  fun `add a document with a virus returns bad request with body`() {
+  fun `add documents which all fail virus scan returns bad request with fileErrors list of objects wrapping category, filename and error text as body`() {
     expectAVirusWillBeFound()
 
     val recall = authenticatedClient.bookRecall(bookRecallRequest)
 
     val response = authenticatedClient.partBRecord(
       recall.recallId,
       partBRequest(),
-      HttpStatus.BAD_REQUEST,
-      ErrorResponse::class.java
+      BAD_REQUEST
+    ).expectBody(MultiFileErrorResponse::class.java).returnResult().responseBody!!
+
+    assertThat(
+      response,
+      equalTo(
+        MultiFileErrorResponse(
+          BAD_REQUEST,
+          listOf(
+            FileError(PART_B_RISK_REPORT, partBFileName, "VirusFoundException"),
+            FileError(PART_B_EMAIL_FROM_PROBATION, emailFileName, "VirusFoundException"),
+            FileError(OASYS_RISK_ASSESSMENT, oasysFileName, "VirusFoundException")
+          )
+        )
+      )
     )
+  }
 
-    assertThat(response, equalTo(ErrorResponse(HttpStatus.BAD_REQUEST, "VirusFoundException")))
+  @Test
+  fun `add documents which all fail virus scan returns bad request with json array fileErrors of objects of category, filename and error text as body`() {
+    expectAVirusWillBeFound()
+
+    val recall = authenticatedClient.bookRecall(bookRecallRequest)
+
+    authenticatedClient.partBRecord(
+      recall.recallId,
+      partBRequest(),
+      BAD_REQUEST
+    ).expectBody()
+      .jsonPath("$.fileErrors.length()").isEqualTo(3)
+      .jsonPath("$.fileErrors[0].category").isEqualTo(PART_B_RISK_REPORT.toString())
+      .jsonPath("$.fileErrors[0].fileName").isEqualTo(partBFileName.value)
+      .jsonPath("$.fileErrors[0].error").isEqualTo("VirusFoundException")
+      .jsonPath("$.fileErrors[1].category").isEqualTo(PART_B_EMAIL_FROM_PROBATION.toString())
+      .jsonPath("$.fileErrors[2].category").isEqualTo(OASYS_RISK_ASSESSMENT.toString())
   }
```

Comment:
Do we have a partial virus scan error?  e.g. 1/3 fail? Also should these tests assert that the uploaded docs aren't saved/added to the recall?

Reply:
There is a 1 of 3 error test in `PartBRecordServiceTest` at `Returns failure if document scan returns Failure`.  
This also addresses your question: "Feel like also missing a test on PartBRecordServiceTest?"
I thought about a parameterised test to prove each individually fails, but then where do you stop?

Reply:
There is a 1 of 3 error test in `PartBRecordServiceTest` at `Returns failure if document scan returns Failure`.  
This also addresses your question: "Feel like also missing a test on PartBRecordServiceTest?"
I thought about a parameterised test to prove each individually fails, but then where do you stop?

Reply:
I think cos its not in this PR I must have missed it.